### ISSUE TITLE
 create custom Config  fails  with `--extra-template-data` against the default pydantic template

### ISSUE THREAD

**Issue by auphofBSF** [koxudaxi/datamodel-code-generator#860]
**Describe the bug**
`--extra-template-data` to create custom `class Config` fails when using the  default accompanying pydantic template as at current commit  last PR #859 

I have created a test for this and included a fix in PR #861, This PR however this also highlighted 1 further issue.

1) The accompanying pydantic template, does not have the "comment" templates part as in the test template, 
✅ Fixed by adding to the default template as shown in the test template

The new test  in PR handles `--extra-template-data` when  create custom `class Config` correctly

Redacted dump of error
```bash
Exception has occurred: TypeError
'dict' object is not callable
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\model\template\pydantic\Config.jinja2", line 13, in root
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\model\template\pydantic\BaseModel.jinja2", line 50, in root
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\model\base.py", line 145, in _render
    return self.template.render(*args, **kwargs)
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\model\base.py", line 286, in render
    response = self._render(
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\model\base.py", line 148, in __str__
    return self.render()
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\parser\base.py", line 76, in <genexpr>
    return '\n\n\n'.join(str(m) for m in templates)
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\parser\base.py", line 76, in dump_templates
    return '\n\n\n'.join(str(m) for m in templates)
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\parser\base.py", line 694, in parse
    code = dump_templates(models)
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\__init__.py", line 369, in generate
    results = parser.parse()
  File "<REDACTED_PATH>\datamodel-code-generator\datamodel_code_generator\__main__.py", line 550, in main
    generate(
  File "<REDACTED_PATH>\datamodel-code-generator\tests\test_main.py", line 526, in test_main_extra_template_data_config
    main(
```

**To Reproduce**

Example `template_data/extra.json`:
```json

{
  "#all#": {
    "config":{"arbitrary_types_allowed": "True"}
  }
}


```

Used commandline:
```
$ datamodel-codegen --input <SOURCE_MODEL> --output <OUTPUTMODEL>  --extra-template-data ./template_data/extra.json
```

**Expected behavior**
The output model should have appropriate `Class Config` lines create

**Version:**
 - OS: Windows 11
 - Python version: ^3.9
 - datamodel-code-generator version: 0.13.1

**Additional context**

**Comment by auphofBSF** [comment]
Initial identification of a possible further issue with Field descriptions in the first issues submission were incorrect interpretations of the capture in the test development. This is removed from above description
This has been fixed with a second commit and the ADDITIONAL test ` test_main_extra_template_data_config` passes 

There should be no outstanding issue other than documentation , which is really lacking in this area see long standing #184 and associated chain. 
This test case could be an example for custom config
I am glad I persevered to get it working because it enables great customization and got me understand the custom template mechanism. Thank you @koxudaxi  for this overall superb model generator
